api/integrations/github/models.py

```diff
@@ -57,21 +64,46 @@ class Meta:
                     "repository_name",
                 ],
                 name="unique_repository_data",
+                condition=models.Q(deleted_at__isnull=True),
             )
         ]
+        ordering = ("id",)
 
     @hook(BEFORE_DELETE)
     def delete_feature_external_resources(
         self,
     ) -> None:
         from features.feature_external_resources.models import (
             FeatureExternalResource,
+            ResourceType,
         )
 
         FeatureExternalResource.objects.filter(
             feature_id__in=self.project.features.values_list("id", flat=True),
             type__in=[
-                FeatureExternalResource.ResourceType.GITHUB_ISSUE,
-                FeatureExternalResource.ResourceType.GITHUB_PR,
+                ResourceType.GITHUB_ISSUE,
+                ResourceType.GITHUB_PR,
             ],
+            # Filter by url containing the repository owner and name
+            url__contains=f"{self.repository_owner}/{self.repository_name}",
         ).delete()
+
+    @hook(AFTER_CREATE)
+    def create_github_tags(
+        self,
+    ) -> None:
+        from integrations.github.constants import (
+            GitHubTag,
+            github_tag_description,
+        )

```

Comment:
Are you sure you need to inline import these constants? Surely they would be topline imports since there should be no cyclical import issues.

Reply:
Yes. I have a circular import issue otherwise.